src/device.rs

```diff
@@ -53,11 +54,15 @@ impl Device {
     }
 
     pub fn set_key(&self, index_: u32, keyval: u32, modifiers: ::ModifierType) {
-        unsafe { ffi::gdk_device_set_key(self.to_glib_none().0, index_, keyval, modifiers) }
+        unsafe { ffi::gdk_device_set_key(self.to_glib_none().0, index_, keyval, modifiers.to_glib()) }
     }
 
-    pub fn get_key(&self, index_: u32, keyval: &mut u32, modifiers: &mut ::ModifierType) -> bool {
-        unsafe { from_glib(ffi::gdk_device_get_key(self.to_glib_none().0, index_, keyval, modifiers)) }
+    pub fn get_key(&self, index_: u32, keyval: &mut u32) -> (bool,::ModifierType) {
+        unsafe {
+          let mut modifiers = mem::uninitialized();
+          let ret = from_glib(ffi::gdk_device_get_key(self.to_glib_none().0, index_, keyval, &mut modifiers));
```

Comment:
This looks like a typical case of out-params + boolean, returning `Option<(u32, ModifierType)>`.